projects/plugins/crm/includes/ZeroBSCRM.Permissions.php

```diff
@@ -577,6 +577,9 @@ function zeroBSCRM_permsObjType( $obj_type_id = -1 ) { // phpcs:ignore WordPress
 
 		case ZBS_TYPE_TASK:
 			return zeroBSCRM_perms_tasks();
+
+		case ZBS_TYPE_SEGMENT:
+			return jpcrm_perms_segments();
```

Comment:
This works. Since we're calling this the same as the contacts permission, another option would be to collapse this into the `ZBS_TYPE_CONTACT` case block.

Reply:
It seems the function used in the ZBS_TYPE_CONTACTS case is deprecated - `zeroBSCRM_permsCustomers`, but replacing it with `jpcrm_can_user_manage_contacts` is a little messier as we need the contact id, contact etc. So it'd either be changing the segment to use the deprecated function or using a new function. It looks like there a fair few usages of the deprecated function, possibly for the same reason (needing more data to pass in to the function).
I moved it. Commit: dace982a8c724ea2209ac7309620c76b72bb5f0c

Reply:
Yeah, the deprecation was a bit controversial; we can ignore it in this case.